cloud/data_source_pulsar_cluster.go

```diff
@@ -366,6 +367,10 @@ func dataSourcePulsarClusterRead(ctx context.Context, d *schema.ResourceData, me
 	if releaseChannel != "" {
 		_ = d.Set("release_channel", releaseChannel)
 	}
+
+	if instanceName != "" {
+		_ = d.Set("instance_name", instanceName)
```

Comment:
This is effectively trying to read and write to the same field, resulting in a null value. Should instead be `d.Set("instance_name", pulsarCluster.Spec.InstanceName)` or probably `pulsarInstance.Spec.Name`